classes/PaymentModule.php

```diff
@@ -1076,7 +1076,12 @@ protected function createOrderCartRules(
             //  This is an "amount" reduction, not a reduction in % or a gift
             // THEN
             //  The voucher is cloned with a new value corresponding to the remainder
-            if (count($order_list) == 1 && $values['tax_incl'] > ($order->total_products_wt - $total_reduction_value_ti) && $cartRule->partial_use == 1 && $cartRule->reduction_amount > 0) {
+            if ($cartRule->reduction_tax) {
+                $remainingValue = $cartRule->reduction_amount - $values['tax_incl'];
```

Comment:
`$remainingValue = $cartRule->reduction_amount - $values[$cartRule->reduction_tax ? 'tax_incl': 'tax_excl'];`
Otherwise we will miss something one day :D